x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/result_settings.tsx

```diff
@@ -59,36 +70,65 @@ export const ResultSettings: React.FC<Props> = ({ engineBreadcrumb }) => {
           'xpack.enterpriseSearch.appSearch.engine.resultSettings.pageDescription',
           { defaultMessage: 'Enrich search results and select which fields will appear.' }
         )}
-        rightSideItems={[
-          <EuiButton
-            data-test-subj="SaveResultSettings"
-            color="primary"
-            fill
-            onClick={saveResultSettings}
-          >
-            {SAVE_BUTTON_LABEL}
-          </EuiButton>,
-          <EuiButton
-            data-test-subj="ResetResultSettings"
-            color="danger"
-            onClick={confirmResetAllFields}
-          >
-            {RESTORE_DEFAULTS_BUTTON_LABEL}
-          </EuiButton>,
-          <EuiButtonEmpty data-test-subj="ClearResultSettings" onClick={clearAllFields}>
-            {CLEAR_BUTTON_LABEL}
-          </EuiButtonEmpty>,
-        ]}
+        rightSideItems={
+          hasSchema
```

Comment:
Nice! I was going to ask if we should do an early return like the Loading screen but I think this is good as well.